plugins/teiid-8.11/src/main/java/org/komodo/teiid/JbossExtensions.java

```diff
@@ -0,0 +1,236 @@
+        if (!isDomainMode(admin))
+            return null;
+
+        return getChildNodeNames(admin, null, PROFILE).get(0);
+    }
+
+    private void addProfileNode(Admin admin, DefaultOperationRequestBuilder builder) throws Exception {
+        if (! isDomainMode(admin))
+            return;
+
+        String profile = getProfileName(admin);
+        if (profile != null)
+            builder.addNode(PROFILE, profile);
+    }
+
+    /**
+     * @param admin
+     * @param subsystem
+     * @param operationName
+     * @param params
+     * @return a new model node request
+     * @throws Exception
+     */
+    private ModelNode buildRequest(Admin admin, String subsystem, String operationName, String... params) throws Exception {
+        DefaultOperationRequestBuilder builder = new DefaultOperationRequestBuilder();
+        final ModelNode request;
+        try {
+            if (subsystem != null) {
+                addProfileNode(admin, builder);
+                builder.addNode(SUBSYSTEM, subsystem); //$NON-NLS-1$
+            }
+            builder.setOperationName(operationName);
+            request = builder.buildRequest();
+            if (params != null && params.length % 2 == 0) {
+                for (int i = 0; i < params.length; i+=2) {
+                    builder.addProperty(params[i], params[i+1]);
+                }
+            }
+        } catch (Exception e) {
+            throw new Exception(Messages.getString(Messages.TeiidInstance.buildOperationFailure, operationName), e); //$NON-NLS-1$
+        }
+
+        return request;
+    }
+
+    /**
+     * @param operationResult
+     * @param mapper
+     * @return list of results from operation
+     */
+    private <T> List<T> getList(ModelNode operationResult,  MetadataMapper<T> mapper) {
+        if(!operationResult.hasDefined(RESULT)) {
+            return Collections.emptyList();
+        }
+
+        List<ModelNode> nodeList = operationResult.get(RESULT).asList(); //$NON-NLS-1$
+        if(nodeList.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        List<T> list = new ArrayList<T>(nodeList.size());
+        for(ModelNode node : nodeList) {
+            list.add(mapper.unwrap(node));
+        }
+        return list;
+    }
+
+    /**
+     * @param admin
+     * @return the drivers available on the underlying jboss server
+     * @throws Exception
+     */
+    public Collection<DataSourceDriver> getDataSourceDrivers(Admin admin) throws Exception {
+        if (admin == null)
+            return Collections.emptyList();
+
+        Set<DataSourceDriver> dataSourceDrivers = new HashSet<DataSourceDriver>();
+
+        if (! isDomainMode(admin)) {
+            //'installed-driver-list' not available in the domain mode.
+            final ModelNode request = buildRequest(admin, DATA_SOURCES, INSTALLED_DRIVERS_LIST);
+            ModelNode outcome = getConnection(admin).execute(request);
+            if (Util.isSuccess(outcome)) {
```

Comment:
Do we need to handle when the outcome is not successful? Unless an unsuccessful outcome could mean there are no drivers, I'm wondering if the callers should know the difference between not having any drivers and there was a problem obtaining the drivers.